src/kakarot/accounts/library.cairo

```diff
@@ -684,6 +684,7 @@ namespace Internals {
 
         jmp cond if remaining_bytes != 0;
 
+        assert value = 0;
```

Comment:
my bad in the PR description, as per zellic comment we need to do this at the end of each chunk, not at the end of the whole read. So missing here as well

```
        cond:
        jmp body if count != 0;
        assert value = 0;
        jmp read;
```